### ISSUE TITLE
vCenter -  vRouter Solution

### ISSUE THREAD

**Issue by ccesario** [OpenNebula/one#2334]
**Description**
I would interesting provide vCenter vRouter managed solution.

**Use case**
Using API to crontol vRouter it is possible create DHCP pool, VPN solution, Interface management, Zones, Firewall Rules, Dynamic routing, VLANs.

**Interface Changes**
Sunstone, API changes

**Additional Context**
Possible solutions
 **vyos**  :: https://vyos.io
    - https://github.com/vyos/python-vyos-mgmt
    - https://blog.vyos.net/vyos-2-dot-0-development-digest-number-9-socket-communication-functionality-complete-parser-and-open-tasks
     - https://wiki.vyos.net/wiki/Perl_API
     - https://docs.ansible.com/ansible/2.4/vyos_config_module.html
     - http://blog.vyos.net/making-first-boot-scripts-just-got-easier-but-building-vyos-1x-got-a-bit-harder

 **opnsense**  :: https://opnsense.org
    - https://docs.opnsense.org/development/how-tos/api.html
    - https://docs.opnsense.org/development/api.html
    - https://comsystoreply.de/blog-post/private-cloud-infrastructure-with-opennebula  (using opnsense as router)

## Progress Status
- [ ] Branch created 
- [ ] Code committed to development branch
- [ ] Testing - QA
- [ ] Documentation 
- [ ] Release notes - resolved issues, compatibility, known issues
- [ ] Code committed to upstream release/hotfix branches
- [ ] Documentation committed to upstream release/hotfix branches

**Comment by ccesario** [comment]
More content that canal help
http://blog.vyos.net/vyos-remote-management-library-for-python

**Comment by tinova** [comment]
Evaluate if this is covered by NSX-v integration in #3440, or with the WIP NFV appliance.

**Comment by ospalax** [comment]
Creating DHCP pool by VNF appliance contextualization is possible now. Interface and VLAN managment is provided by OpenNebula either via dynamic attach/detach of NICs or other means(?) at least on the physical level - IPs, networks etc. is again by contextualization.

If by zones is meant DNS zone files then no - VNF appliance does not support custom domain configuration - it has DNS service only as a recursor.

Dynamic routing via special protocol like BGP is also not supported by VNF appliance - that could possibly be added in the future.

VPN is also not included in the VNF appliance currently but this could be also added (but which one? IPsec? OpenVPN? WireGuard? other? all?)

Firewall is not supported and frankly the only support I would say is meaningful is just a way to provide one context variable where to put complete file with all iptables rules...I see no reason to implement yet another iptable wrapper and reinvent wheel like shorewall, firewalld, ufw etc.

TL;DR VNF appliance does all its (re)configuration through contextulization scripts which are triggered by OpenNebula - there is no service running inside and providing API and (re)configuration on-the-fly.

**Comment by tinova** [comment]
I think the overlap is enough to discard vCenter VR integration, please let us know if there is any use case not covered by the VNF appliance.

@ospalax my money is on WireGuard ;)
